src/App.tsx

```diff
@@ -10,31 +10,33 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppSelector } from './app/hooks';
+import { useDispatch } from 'react-redux';
+import { setError, setLoaded, setPosts } from './features/posts';
+import { setSelectedPost } from './features/selectedPost';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const { items, loaded, hasError } = useAppSelector(state => state.posts);
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const author = useAppSelector(state => state.author);
+  const selectedPost = useAppSelector(state => state.selectedPost);
+
+  const dispatch = useDispatch();
 
   function loadUserPosts(userId: number) {
-    setLoaded(false);
+    dispatch(setLoaded(false));
 
     getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
+      .then(posts => dispatch(setPosts(posts)))
+      .catch(() => dispatch(setError(true)))
       // We disable the spinner in any case
-      .finally(() => setLoaded(true));
+      .finally(() => dispatch(setLoaded(true)));
   }
```

Comment:
Consider handling errors more gracefully. Instead of just setting the error state to true when an error occurs, it would be a good idea to also log the error or show a user-friendly error message. This way, you can debug issues more easily and improve the user experience.